Source/Turbo Navigator/TurboNavigationHierarchyController.swift

```diff
@@ -86,7 +86,7 @@ class TurboNavigationHierarchyController {
                 delegate.visit(visitable, on: .main, with: proposal.options)
             }
         case .modal:
-            if navigationController.presentedViewController != nil, !modalNavigationController.isBeingDismissed {
+            if modalNavigationController.presentingViewController != nil, !modalNavigationController.isBeingDismissed {
                 pushOrReplace(on: modalNavigationController, with: controller, via: proposal)
             } else {
                 modalNavigationController.setViewControllers([controller], animated: proposal.animated)
```

Comment:
This change will work correctly for the scenario outlined in the description. However, the prerequisite is that the alert/modal presented on the root VC has been dismissed, otherwise we'll get a warning - "Attempt to present <modalNavigationController>  on <navigationController>  which is already presenting <some view controller>."

We could handle this in Turbo by dismissing any view controller presented on the `navigationController` before presenting the `modalNavigationController`.

On the other hand, I'm wondering if manually altering the Turbo stack is a case of misuse.

Reply:
> We could handle this in Turbo by dismissing any view controller presented on the navigationController before presenting the modalNavigationController.

I think this is the best approach. I'll work on it.

> On the other hand, I'm wondering if manually altering the Turbo stack from the outside is a case of misuse.

I mostly agree 🤔. Having said that, I could see how presenting an alert (or a custom toast) as a consequence of a native process would be something ephemeral that doesn't necessarily have to go through the path config. Let me dismiss if a push or replace is happening and see how that feels then we can come back to this.